optimism/src/mips/witness.rs

```diff
@@ -616,6 +622,18 @@ impl<Fp: Field> InterpreterEnv for Env<Fp> {
         if self.preimage_bytes_read.unwrap() == preimage_len as u64 {
             let mut keccak_env = KeccakEnv::<Fp>::default();
             keccak_env.hash(self.preimage.as_ref().unwrap());
+            match self.preimage_key {
+                Some(preimage_key) => {
+                    let bytes31 = (1..32).fold(Fp::zero(), |acc, i| {
+                        acc * Fp::two_pow(8) + Fp::from(preimage_key[i])
+                    });
+                    keccak_env.add_lookup(Lookup::read_one(
```

Comment:
Note: we need to enforce the order the preimage key, i.e. which keccak hash is referring it to.

```
i + r * keccak preimage key
```

Reply:
the hash index, correct